### PR TITLE
Migrates todo-mvp-kotlin to androidX

### PR DESCRIPTION
**Pull request by JoseAlcerreca**
Merges changes in todo-mvp too

### CODE REVIEW

File: todoapp/app/build.gradle [hunk 1]
```diff
@@ -66,43 +65,48 @@ android {
...
-    implementation "com.android.support.test.espresso:espresso-idling-resource:$rootProject.espressoVersion"
-    implementation "android.arch.persistence.room:runtime:$rootProject.roomVersion"
-    kapt "android.arch.persistence.room:compiler:$rootProject.roomVersion"
+    // App dependencies
+    implementation "androidx.appcompat:appcompat:$appCompatVersion"
+    implementation "androidx.cardview:cardview:$cardVersion"
+    implementation "com.google.android.material:material:$materialVersion"
+    implementation "androidx.recyclerview:recyclerview:$recyclerViewVersion"
+    implementation "androidx.annotation:annotation:$androidXAnnotations"
+    implementation "com.google.code.findbugs:jsr305:3.0.2"
```
Comment (nickbutcher) [inline todoapp/app/build.gradle]: Factor version out per other deps?

File: todoapp/app/build.gradle [hunk 2]
```diff
@@ -66,43 +65,48 @@ android {
...
 
     // Dependencies for local unit tests
     testImplementation "junit:junit:$rootProject.ext.junitVersion"
-    testImplementation "org.hamcrest:hamcrest-all:$rootProject.ext.hamcrestVersion"
     testImplementation "org.mockito:mockito-core:$rootProject.ext.mockitoVersion"
-
-    // Android Testing Support Library's runner and rules
-    androidTestImplementation "com.android.support.test:runner:$rootProject.ext.runnerVersion"
-    androidTestImplementation "com.android.support.test:rules:$rootProject.ext.runnerVersion"
+    testImplementation "org.hamcrest:hamcrest-all:$rootProject.ext.hamcrestVersion"
```
Comment (nickbutcher) [inline todoapp/app/build.gradle]: Why do some versions have `rootProject.ext.` prefixes and others do not? Can we drop this to be consistent?

File: todoapp/app/build.gradle (hunk 1 above)
Comment (JoseAlcerreca) [reply todoapp/app/build.gradle]: It's not even used... removed

File: todoapp/app/build.gradle (hunk 2 above)
Comment (JoseAlcerreca) [reply todoapp/app/build.gradle]: Made a mental note in my head to fix it...

**Comment by JoseAlcerreca** [comment]
Circle fails because I did the PR from my project. I'll merge now and fix if there are any problems.

**Pull request merged**